S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java

```diff
@@ -357,12 +366,52 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
+
+    private void  requestPermission()
+    {
+        //ask for the permission in android M
+        int permission = ContextCompat.checkSelfPermission(this,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE);
+
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            Log.i(TAG, "Permission to record denied");
```

Comment:
change to: "Permission to store data in external storage is not granted."

Reply:
done !